### ISSUE TITLE
All-default Any value content becomes undefined after decoded

### ISSUE THREAD

**Issue by 5argon** [stephenh/ts-proto#241]
When using generated `Any` encoder code, with protobuf bytes representing all default values of an another protobuf type, the `Any` encoder will produce a buffer where if decoded with `Any` decoder back, produces only `{ typeUrl: string }` with no `value` field. Understandable, since protobuf treats default field as no data.

If I decided to encode this decoded `Any` back again (as-is, or I nest this `Any` inside other protobuf type and encode that so it in turn need to encode `Any`), now it would produces error since there is a line that checks for `length` of `value` but `value` is undefined.

```js
	encode(message: Any, writer: _m0.Writer = _m0.Writer.create()): _m0.Writer {
		if (message.typeUrl !== '') {
			writer.uint32(10).string(message.typeUrl)
		}
		if (message.value.length !== 0) {
			writer.uint32(18).bytes(message.value)
		}
		return writer
	},
```

Supposed that I edit in `message.value !== undefined` in the encoder to keep the `undefined` value, the `undefined` value can't be used with the intended non-Any decoder (of the type specified on `typeUrl`) anyways since the generated line

```
let end = length === undefined ? reader.len : reader.pos + length
```

Would cause this error :

```
     TypeError: Cannot read property 'len' of undefined
      at Object.decode ((omitted)/test_proto.ts:125:43)
```

Past this, `while(reader.pos)` also need the input to be defined.

So an another fix that check `if (input === undefined)` and early `return baseFooBar as FooBar` immediately is needed, since the input is undefined due to all-default. 

In summary : 
- It seems like the generated `Any` definition `{ typeUrl: string, value: Uint8Array }` isn't right. Maybe it should be `value: Uint8Array | undefined` to support all-default data?
- Non-Any generated decode code need to account for undefined input and return all-default?

**Comment by stephenh** [comment]
Hey, thanks for the report!

I'm not really surprised `Any` has some rough edges, I haven't personally used it yet and there are also aren't any test cases that cover scenarios like you're mentioning.

As a start, could you submit a PR to make like an `integration/any` directory with a minimalist `*.proto` file that reproduces the issue?

If you wanted to go further and just ahead and have the PR fix it, that'd be great too! :-) 

it looks like the `only-types` test includes `Any`:

https://github.com/stephenh/ts-proto/blob/c87b1571972aed875fa724e820e0a7634a38304e/integration/only-types/google/protobuf/any.ts

But it purposefully doesn't cover the encode/decode methods. But you could use that as a place to copy/paste from if necessary.
